### ISSUE TITLE
Regent: Fail to index space launch field polymorphic tasks with projection functor.

### ISSUE THREAD

**Issue by syamajala** [StanfordLegion/legion#783]
```
import "regent"

struct Fields
{
  A : int
  B : int
}

task leaf(r : region(ispace(int3d), int))
where
  reads(r)
do
end

task main()
  var r = region(ispace(int3d, int3d{10, 10, 10}), Fields)
  var p = partition(equal, r, ispace(int3d, int3d{2, 1, 1}))
  fill(r.{A, B}, 1)

  __demand(__index_launch)
  for color in p.colors do
    var pc = (color + int3d{1, 0, 0}) % int3d{2, 1, 1}
    leaf(p[pc].{A})
  end
end

regentlib.start(main)
```

Running the above example results in the following:
```
...dev/legion/language/src/regent/optimize_index_launches.t:476: assertion failed!
stack traceback:
        [C]: in function 'assert'
        ...dev/legion/language/src/regent/optimize_index_launches.t:476: in function 'analyze_noninterference_self'
        ...dev/legion/language/src/regent/optimize_index_launches.t:1135: in function 'optimize_loop_body'
        ...dev/legion/language/src/regent/optimize_index_launches.t:1231: in function 'map_node_postorder'
        /home/seshu/dev/legion/language/src/common/ast.t:520: in function 'map_node_postorder'
        /home/seshu/dev/legion/language/src/common/ast.t:511: in function 'map_node_postorder'
        /home/seshu/dev/legion/language/src/common/ast.t:511: in function </home/seshu/dev/legion/language/src/common/ast.t:505>
        /home/seshu/dev/legion/language/src/regent/passes_hooks.t:41: in function 'optimize'
        /home/seshu/dev/legion/language/src/regent/passes.t:52: in function </home/seshu/dev/legion/language/src/regent/passes.t:48>
        index_space_bug.rg:19: in main chunk
```

**Comment by magnatelee** [comment]
This should be fixed by d134818e. @syamajala Can you pull and try again?

**Comment by syamajala** [comment]
It works.
